### ISSUE TITLE
test failures with Go 1.15

### ISSUE THREAD

**Issue by refola** [syncthing/syncthing#6889]
# Steps
1. Start at Syncthing source folder, with below software setup
2. Run `go run build.go -no-upgrade test`
3. Wait a few minutes

# Result
```
panic: qtls.ClientSessionState not compatible with tls.ClientSessionState

goroutine 1 [running]:
github.com/lucas-clemente/quic-go/internal/handshake.init.2()
        $GOPATH/pkg/mod/github.com/lucas-clemente/quic-go@v0.17.3/internal/handshake/unsafe.go:26 +0x569
FAIL    github.com/syncthing/syncthing/lib/api  0.053s
ok      github.com/syncthing/syncthing/lib/api/auto     (cached)
ok      github.com/syncthing/syncthing/lib/assets       (cached)
ok      github.com/syncthing/syncthing/lib/beacon       (cached)
ok      github.com/syncthing/syncthing/lib/build        (cached)
ok      github.com/syncthing/syncthing/lib/config       0.167s
panic: qtls.ClientSessionState not compatible with tls.ClientSessionState

goroutine 1 [running]:
github.com/lucas-clemente/quic-go/internal/handshake.init.2()
        $GOPATH/pkg/mod/github.com/lucas-clemente/quic-go@v0.17.3/internal/handshake/unsafe.go:26 +0x569
FAIL    github.com/syncthing/syncthing/lib/connections  0.025s
ok      github.com/syncthing/syncthing/lib/connections/registry (cached)
ok      github.com/syncthing/syncthing/lib/db   (cached)
ok      github.com/syncthing/syncthing/lib/db/backend   (cached)
ok      github.com/syncthing/syncthing/lib/dialer       (cached) [no tests to run]
ok      github.com/syncthing/syncthing/lib/discover     (cached)
ok      github.com/syncthing/syncthing/lib/events       (cached)
ok      github.com/syncthing/syncthing/lib/fs   24.164s
ok      github.com/syncthing/syncthing/lib/ignore       (cached)
?       github.com/syncthing/syncthing/lib/locations    [no test files]
ok      github.com/syncthing/syncthing/lib/logger       (cached)
panic: qtls.ClientSessionState not compatible with tls.ClientSessionState

goroutine 1 [running]:
github.com/lucas-clemente/quic-go/internal/handshake.init.2()
        $GOPATH/pkg/mod/github.com/lucas-clemente/quic-go@v0.17.3/internal/handshake/unsafe.go:26 +0x569
FAIL    github.com/syncthing/syncthing/lib/model        0.023s
ok      github.com/syncthing/syncthing/lib/nat  (cached)
ok      github.com/syncthing/syncthing/lib/osutil       0.269s
ok      github.com/syncthing/syncthing/lib/pmp  (cached) [no tests to run]
ok      github.com/syncthing/syncthing/lib/protocol     (cached)
ok      github.com/syncthing/syncthing/lib/rand (cached)
?       github.com/syncthing/syncthing/lib/rc   [no test files]
ok      github.com/syncthing/syncthing/lib/relay/client (cached) [no tests to run]
ok      github.com/syncthing/syncthing/lib/relay/protocol       (cached) [no tests to run]
ok      github.com/syncthing/syncthing/lib/scanner      5.540s
?       github.com/syncthing/syncthing/lib/sha256       [no test files]
ok      github.com/syncthing/syncthing/lib/signature    (cached)
ok      github.com/syncthing/syncthing/lib/stats        (cached) [no tests to run]
?       github.com/syncthing/syncthing/lib/stun [no test files]
ok      github.com/syncthing/syncthing/lib/sync (cached)
panic: qtls.ClientSessionState not compatible with tls.ClientSessionState

goroutine 1 [running]:
github.com/lucas-clemente/quic-go/internal/handshake.init.2()
        $GOPATH/pkg/mod/github.com/lucas-clemente/quic-go@v0.17.3/internal/handshake/unsafe.go:26 +0x569
FAIL    github.com/syncthing/syncthing/lib/syncthing    0.019s
?       github.com/syncthing/syncthing/lib/testutils    [no test files]
ok      github.com/syncthing/syncthing/lib/tlsutil      (cached)
ok      github.com/syncthing/syncthing/lib/upgrade      (cached)
ok      github.com/syncthing/syncthing/lib/upnp (cached)
?       github.com/syncthing/syncthing/lib/ur   [no test files]
ok      github.com/syncthing/syncthing/lib/ur/contract  (cached)
ok      github.com/syncthing/syncthing/lib/util (cached)
ok      github.com/syncthing/syncthing/lib/versioner    0.052s
ok      github.com/syncthing/syncthing/lib/watchaggregator      (cached)
ok      github.com/syncthing/syncthing/lib/weakhash     (cached)
?       github.com/syncthing/syncthing/cmd/stcli        [no test files]
?       github.com/syncthing/syncthing/cmd/stcompdirs   [no test files]
ok      github.com/syncthing/syncthing/cmd/stcrashreceiver      (cached)
?       github.com/syncthing/syncthing/cmd/stdisco      [no test files]
ok      github.com/syncthing/syncthing/cmd/stdiscosrv   0.072s
?       github.com/syncthing/syncthing/cmd/stevents     [no test files]
?       github.com/syncthing/syncthing/cmd/stfileinfo   [no test files]
?       github.com/syncthing/syncthing/cmd/stfinddevice [no test files]
?       github.com/syncthing/syncthing/cmd/stfindignored        [no test files]
?       github.com/syncthing/syncthing/cmd/stgenfiles   [no test files]
?       github.com/syncthing/syncthing/cmd/stindex      [no test files]
ok      github.com/syncthing/syncthing/cmd/strelaypoolsrv       (cached)
?       github.com/syncthing/syncthing/cmd/strelaypoolsrv/auto  [no test files]
?       github.com/syncthing/syncthing/cmd/strelaysrv   [no test files]
?       github.com/syncthing/syncthing/cmd/strelaysrv/testutil  [no test files]
?       github.com/syncthing/syncthing/cmd/stsigtool    [no test files]
?       github.com/syncthing/syncthing/cmd/stupgrades   [no test files]
?       github.com/syncthing/syncthing/cmd/stvanity     [no test files]
?       github.com/syncthing/syncthing/cmd/stwatchfile  [no test files]
panic: qtls.ClientSessionState not compatible with tls.ClientSessionState

goroutine 1 [running]:
github.com/lucas-clemente/quic-go/internal/handshake.init.2()
        $GOPATH/pkg/mod/github.com/lucas-clemente/quic-go@v0.17.3/internal/handshake/unsafe.go:26 +0x569
FAIL    github.com/syncthing/syncthing/cmd/syncthing    0.021s
?       github.com/syncthing/syncthing/cmd/uraggregate  [no test files]
ok      github.com/syncthing/syncthing/cmd/ursrv        (cached)
FAIL
exit status 1
exit status 1
```
(`$GOPATH` environment variable name substituted for actual path)

# Expectation
All tests should complete successfully, as they do with Go 1.14.7.

# Software setup
- Syncthing version: v1.8.0, from the source release
- Go version: 1.15
- OS: Chakra Linux (Arch-like)

**Comment by AudriusButkevicius** [comment]
We ourselves are currently not building using 1.15, and are blocked in doing so by https://github.com/lucas-clemente/quic-go/issues/2614

**Comment by calmh** [comment]
You have to pass -tags noquic to avoid building with the quic package that doesn't support go 1.15

**Comment by jiftle** [comment]
me too.

**Comment by jiftle** [comment]
me too.

**Comment by AudriusButkevicius** [comment]
Its fixed on master but it has to be 1.15 release and not some rc.
